Automattic/jetpack
# Allow universal analytics (analytics.js) to be used without WooCommerce

Fixes #9129

#### Changes proposed in this Pull Request:

* Adds a filter to allow end users to force-activate Universal Analytics (analytics.js) even if WooCommerce is not present, e.g.

`add_filter( 'jetpack_wga_use_universal_analytics', '__return_true' );`

* Initializes singleton in init action to give time for plugins/themes to use that filter

#### Testing instructions:

* Install and connect Jetpack using this branch
* Install WooCommerce
* Enable Google Analytics Enhanced eCommerce Analytics in Jetpack Settings
* View source to ensure that analytics.js is enqueued in the head tag (at/near the top) (including the ec module request) and your eCommerce events (like Add to Cart) appear in your Javascript console Network tab (turn log preservation on)
* Deactivate WooCommerce
* Verify ga.js is enqueued instead (this is the "legacy" analytics JS)
* Add the filter above to force universal analytics on
* View source to ensure that analytics.js is enqueued in the head tag (at/near the top), including the pageview universal command (but not including the ec module command)

cc @joendotcom @georgestephanis @rtrevellyan 

IMPORTANT NOTE: It is the responsibility of the site owner to ensure their site's privacy policy correctly reflects the use of Google Analytics Universal Analytics when this filter is enabled.
